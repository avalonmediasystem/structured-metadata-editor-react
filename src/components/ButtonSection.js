import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/show-forms';
import { ButtonToolbar, Button } from 'react-bootstrap';
import APIUtils from '../api/Utils';
import AlertContainer from '../containers/AlertContainer';

const apiUtils = new APIUtils();

class ButtonSection extends Component {
  handleHeadingClick = () => {
    this.props.showModal('ADD', 'div');
  };

  handleTimeSpanClick = () => {
    this.props.showModal('ADD', 'span');
  };

  handleSaveItClick = () => {
    let postData = { json: this.props.smData[0] };
    apiUtils
      .postRequest('structure.json', postData)
      .then(response => {
        this.props.handleResponse(response.status, response.statusText);
      })
      .catch(error => {
        if (error.response !== undefined) {
          this.props.handleResponse(error.response.status);
        } else {
          this.props.handleResponse(error.request.status);
        }
      });
  };

  render() {
    return (
      <section className="button-section">
        <hr />
        <ButtonToolbar>
          <Button onClick={this.handleHeadingClick}>Add a Heading</Button>
          <Button onClick={this.handleTimeSpanClick}>Add a Timespan</Button>
          <Button onClick={this.handleSaveItClick}>Save Structure</Button>
        </ButtonToolbar>
        <AlertContainer />
      </section>
    );
  }
}

const mapStateToProps = state => ({
  showForms: state.showForms,
  smData: state.smData
});

export default connect(
  mapStateToProps,
  actions
)(ButtonSection);
