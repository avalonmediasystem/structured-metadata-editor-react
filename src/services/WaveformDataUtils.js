import StructuralMetadataUtils from './StructuralMetadataUtils';
import Peaks from 'peaks.js';

const structMetadataUtils = new StructuralMetadataUtils();

// Colors for segments from Avalon branding pallette
const colors = ['#80A590', '#2A5459'];

export default class WaveformDataUtils {
  initPeaks(smData, peaksInstance, options) {
    let initSegments = [];
    let count = 0;

    // Recursively build segments for timespans in the structure
    let createSegment = items => {
      for (let item of items) {
        if (item.type === 'span') {
          count = count > 1 ? 0 : count;
          const { begin, end, label, id } = item;
          initSegments.push({
            startTime: structMetadataUtils.toMs(begin) / 1000,
            endTime: structMetadataUtils.toMs(end) / 1000,
            editable: true,
            labelText: label,
            id: id,
            color: colors[count]
          });
          count++;
        }
        if (item.items && item.items.length > 0) {
          createSegment(item.items);
        }
      }
    };

    // Build segments from initial metadata structure
    createSegment(smData);

    // Initialize peaks instance
    peaksInstance = Peaks.init({
      ...options,
      segments: initSegments
    });

    return peaksInstance;
  }
}
