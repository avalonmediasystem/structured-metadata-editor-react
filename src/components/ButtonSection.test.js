import React from 'react';
import { PureButtonSection } from './ButtonSection';
import { shallow } from 'enzyme';
import { testMetadataStructure } from '../test/TestStructure';
import Peaks from 'peaks';
import WaveformDataUtils from '../services/WaveformDataUtils';

const waveformUtils = new WaveformDataUtils();

describe('ButtonSection component', () => {
  let wrapper, props;
  let options = {
    container: null,
    mediaElement: null,
    dataUri: null,
    dataUriDefaultFormat: 'json',
    keyboard: true,
    _zoomLevelIndex: 0,
    _zoomLevels: [512, 1024, 2048, 4096]
  };
  beforeEach(() => {
    const peaks = { peaks: Peaks.init(options) };
    let initforms = {
      editingDisabled: false,
      structureRetrieved: true,
      waveformRetrieved: true
    };
    props = {
      peaksInstance: peaks,
      smData: testMetadataStructure,
      forms: initforms,
      createTempSegment: jest.fn(() => {
        waveformUtils.insertTempSegment(peaks.peaks);
      }),
      handleEditingTimespans: jest.fn(code => {
        if (code === 0) {
          initforms.editingDisabled = true;
        } else {
          initforms.editingDisabled = false;
        }
      })
    };
    wrapper = shallow(<PureButtonSection {...props} />);
  });

  test('component renders without crashing', () => {
    const {
      headingOpen,
      timespanOpen,
      isInitializing,
      initSegment,
      alertObj,
      disabled
    } = wrapper.instance().state;
    expect(wrapper).toMatchSnapshot();
    expect(headingOpen).toBeFalsy();
    expect(timespanOpen).toBeFalsy();
    expect(isInitializing).toBeTruthy();
    expect(initSegment).toBeNull();
    expect(alertObj).toBeNull();
    expect(disabled).toBeTruthy();
    expect(wrapper.find('Button').at(0)).toBeDefined();
  });

  test('component render null when structureRetrieved & waveformRetrieved are false', () => {
    const newProps = {
      ...props,
      forms: {
        structureRetrieved: false,
        waveformRetrieved: false
      }
    };
    // Set props which re-renders the component
    wrapper.setProps({
      ...newProps
    });
    // Component is null when both structure & waveform master files are not retrieved
    expect(wrapper.type()).toBeNull();
  });

  test('component render null when either structureRetrieved or waveformRetrieved is false', () => {
    const newProps = {
      ...props,
      forms: {
        structureRetrieved: true,
        waveformRetrieved: false
      }
    };
    // Set props which re-renders the component
    wrapper.setProps({
      ...newProps
    });
    // Component is null when both structure & waveform master files are not retrieved
    expect(wrapper.type()).toBeNull();
  });

  test('tests Add a Heading', () => {
    const addheading = wrapper.find('Button').at(0);
    addheading.simulate('click');

    expect(wrapper.instance().props.forms.editingDisabled).toBeTruthy();

    expect(wrapper.instance().state.headingOpen).toBeTruthy();
    expect(wrapper.instance().state.timespanOpen).toBeFalsy();
    expect(wrapper.instance().state.disabled).toBeFalsy();
  });

  test('tests Add a Timespan', () => {
    // Move the playhead to a time within an existing segment
    wrapper.instance().props.peaksInstance.peaks.player.seek(450);

    const addSpan = wrapper.find('Button').at(1);
    addSpan.simulate('click');

    expect(wrapper.instance().props.createTempSegment).toHaveBeenCalled();
    expect(wrapper.instance().props.forms.editingDisabled).toBeTruthy();

    expect(wrapper.instance().state.timespanOpen).toBeTruthy();
    expect(wrapper.instance().state.headingOpen).toBeFalsy();
    expect(wrapper.instance().state.disabled).toBeFalsy();
    expect(wrapper.instance().state.initSegment).toEqual({
      id: 'temp-segment',
      startTime: 480.01,
      endTime: 540.01,
      editable: true,
      color: '#FBB040'
    });
  });

  test('tests Add a Timespan when there is no space to add a new timespan', () => {
    const { peaks } = wrapper.instance().props.peaksInstance;
    // Prep work: add segments to reach the end of the file
    peaks.segments.add([
      {
        startTime: 900.01,
        endTime: 1200.99,
        id: '123a-456b-789c-9d',
        editable: true,
        color: '#FBB040'
      },
      {
        startTime: 1201,
        endTime: 1738.95,
        id: '123a-456b-789c-10d',
        editable: true,
        color: '#FBB040'
      }
    ]);
    // Move the playhead to a time within an existing segment
    peaks.player.seek(1200);
    wrapper
      .find('Button')
      .at(1)
      .simulate('click');

    expect(wrapper.instance().props.forms.editingDisabled).toBeTruthy();

    expect(wrapper.instance().state.disabled).toBeFalsy();
    expect(wrapper.instance().state.alertObj.alertStyle).toEqual('warning');
    expect(wrapper.instance().state.alertObj.message).toEqual(
      'Time ahead has timespans reaching the end of media file, there is no available time to insert a new timespan'
    );
    expect(wrapper.instance().state.timespanOpen).toBeFalsy();
  });
});
