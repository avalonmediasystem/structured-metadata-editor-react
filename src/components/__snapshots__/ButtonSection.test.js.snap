// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSection component component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonSection
    createTempSegment={[MockFunction]}
    handleEditingTimespans={[MockFunction]}
    peaksInstance={
      Object {
        "peaks": Object {
          "options": Object {
            "_zoomLevelIndex": 0,
            "_zoomLevels": Array [
              512,
              1024,
              2048,
              4096,
            ],
            "container": null,
            "dataUri": null,
            "dataUriDefaultFormat": "json",
            "keyboard": true,
            "mediaElement": null,
          },
          "player": Object {
            "_mediaElement": Object {
              "currentTime": 0,
              "duration": 1738.945306,
            },
            "getCurrentTime": [MockFunction],
            "getDuration": [MockFunction],
            "seek": [MockFunction],
          },
          "segments": Object {
            "_peaks": [Circular],
            "_segments": Array [
              Object {
                "color": "#80A590",
                "endTime": 10.32,
                "id": "123a-456b-789c-3d",
                "labelText": "Segment 1.1",
                "startTime": 3.32,
              },
              Object {
                "color": "#2A5459",
                "endTime": 480,
                "id": "123a-456b-789c-4d",
                "labelText": "Segment 1.2",
                "startTime": 11.23,
              },
              Object {
                "color": "#80A590",
                "endTime": 900,
                "id": "123a-456b-789c-8d",
                "labelText": "Segment 2.1",
                "startTime": 543.24,
              },
            ],
            "add": [MockFunction],
            "getSegment": [MockFunction],
            "getSegments": [MockFunction],
            "removeAll": [MockFunction],
            "removeById": [MockFunction],
          },
          "zoom": Object {
            "_peaks": [Circular],
            "_zoomLevelIndex": 2,
            "_zoomLevels": Array [
              512,
              1024,
              2048,
              4096,
            ],
            "zoomIn": [MockFunction],
            "zoomOut": [MockFunction],
          },
        },
      }
    }
    showForms={
      Object {
        "disabled": false,
      }
    }
    smData={
      Array [
        Object {
          "id": "123a-456b-789c-0d",
          "items": Array [
            Object {
              "id": "123a-456b-789c-1d",
              "items": Array [
                Object {
                  "id": "123a-456b-789c-2d",
                  "items": Array [],
                  "label": "Sub-Segment 1.1",
                  "type": "div",
                },
                Object {
                  "begin": "00:00:03.32",
                  "end": "00:00:10.32",
                  "id": "123a-456b-789c-3d",
                  "label": "Segment 1.1",
                  "type": "span",
                },
                Object {
                  "begin": "00:00:11.23",
                  "end": "00:08:00.00",
                  "id": "123a-456b-789c-4d",
                  "label": "Segment 1.2",
                  "type": "span",
                },
              ],
              "label": "First segment",
              "type": "div",
            },
            Object {
              "id": "123a-456b-789c-5d",
              "items": Array [
                Object {
                  "id": "123a-456b-789c-6d",
                  "items": Array [
                    Object {
                      "id": "123a-456b-789c-7d",
                      "items": Array [],
                      "label": "Sub-Segment 2.1.1",
                      "type": "div",
                    },
                    Object {
                      "begin": "00:09:03.24",
                      "end": "00:15:00.00",
                      "id": "123a-456b-789c-8d",
                      "label": "Segment 2.1",
                      "type": "span",
                    },
                  ],
                  "label": "Sub-Segment 2.1",
                  "type": "div",
                },
              ],
              "label": "Second segment",
              "type": "div",
            },
          ],
          "label": "Title",
          "type": "div",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <AlertContainer />,
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            xs={6}
          >
            <Button
              active={false}
              block={true}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
            >
              Add a Heading
            </Button>
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            xs={6}
          >
            <Button
              active={false}
              block={true}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
            >
              Add a Timespan
            </Button>
          </Col>
        </Row>,
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            className="well"
            style={
              Object {
                "marginTop": "1rem",
              }
            }
          >
            <Connect(HeadingFormContainer)
              cancelClick={[Function]}
            />
          </div>
        </Collapse>,
        <Collapse
          appear={false}
          dimension="height"
          getDimensionValue={[Function]}
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <div
            className="well"
            style={
              Object {
                "marginTop": "1rem",
              }
            }
          >
            <Connect(TimespanFormContainer)
              cancelClick={[Function]}
              initSegment={null}
              isInitializing={true}
              timespanOpen={false}
              updateInitialize={[Function]}
            />
          </div>
        </Collapse>,
      ],
      "style": Object {
        "margin": "4rem 0",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": Array [
            <Col
              bsClass="col"
              componentClass="div"
              xs={6}
            >
              <Button
                active={false}
                block={true}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
              >
                Add a Heading
              </Button>
            </Col>,
            <Col
              bsClass="col"
              componentClass="div"
              xs={6}
            >
              <Button
                active={false}
                block={true}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
              >
                Add a Timespan
              </Button>
            </Col>,
          ],
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <Button
                active={false}
                block={true}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
              >
                Add a Heading
              </Button>,
              "componentClass": "div",
              "xs": 6,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": true,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "Add a Heading",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add a Heading",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <Button
                active={false}
                block={true}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
              >
                Add a Timespan
              </Button>,
              "componentClass": "div",
              "xs": 6,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": true,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "Add a Timespan",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add a Timespan",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appear": false,
          "children": <div
            className="well"
            style={
              Object {
                "marginTop": "1rem",
              }
            }
          >
            <Connect(HeadingFormContainer)
              cancelClick={[Function]}
            />
          </div>,
          "dimension": "height",
          "getDimensionValue": [Function],
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(HeadingFormContainer)
              cancelClick={[Function]}
            />,
            "className": "well",
            "style": Object {
              "marginTop": "1rem",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cancelClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appear": false,
          "children": <div
            className="well"
            style={
              Object {
                "marginTop": "1rem",
              }
            }
          >
            <Connect(TimespanFormContainer)
              cancelClick={[Function]}
              initSegment={null}
              isInitializing={true}
              timespanOpen={false}
              updateInitialize={[Function]}
            />
          </div>,
          "dimension": "height",
          "getDimensionValue": [Function],
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(TimespanFormContainer)
              cancelClick={[Function]}
              initSegment={null}
              isInitializing={true}
              timespanOpen={false}
              updateInitialize={[Function]}
            />,
            "className": "well",
            "style": Object {
              "marginTop": "1rem",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cancelClick": [Function],
              "initSegment": null,
              "isInitializing": true,
              "timespanOpen": false,
              "updateInitialize": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AlertContainer />,
          <Row
            bsClass="row"
            componentClass="div"
          >
            <Col
              bsClass="col"
              componentClass="div"
              xs={6}
            >
              <Button
                active={false}
                block={true}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
              >
                Add a Heading
              </Button>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
              xs={6}
            >
              <Button
                active={false}
                block={true}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
              >
                Add a Timespan
              </Button>
            </Col>
          </Row>,
          <Collapse
            appear={false}
            dimension="height"
            getDimensionValue={[Function]}
            in={false}
            mountOnEnter={false}
            timeout={300}
            unmountOnExit={false}
          >
            <div
              className="well"
              style={
                Object {
                  "marginTop": "1rem",
                }
              }
            >
              <Connect(HeadingFormContainer)
                cancelClick={[Function]}
              />
            </div>
          </Collapse>,
          <Collapse
            appear={false}
            dimension="height"
            getDimensionValue={[Function]}
            in={false}
            mountOnEnter={false}
            timeout={300}
            unmountOnExit={false}
          >
            <div
              className="well"
              style={
                Object {
                  "marginTop": "1rem",
                }
              }
            >
              <Connect(TimespanFormContainer)
                cancelClick={[Function]}
                initSegment={null}
                isInitializing={true}
                timespanOpen={false}
                updateInitialize={[Function]}
              />
            </div>
          </Collapse>,
        ],
        "style": Object {
          "margin": "4rem 0",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": Array [
              <Col
                bsClass="col"
                componentClass="div"
                xs={6}
              >
                <Button
                  active={false}
                  block={true}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                >
                  Add a Heading
                </Button>
              </Col>,
              <Col
                bsClass="col"
                componentClass="div"
                xs={6}
              >
                <Button
                  active={false}
                  block={true}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                >
                  Add a Timespan
                </Button>
              </Col>,
            ],
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <Button
                  active={false}
                  block={true}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                >
                  Add a Heading
                </Button>,
                "componentClass": "div",
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": true,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Add a Heading",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Add a Heading",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <Button
                  active={false}
                  block={true}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                >
                  Add a Timespan
                </Button>,
                "componentClass": "div",
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": true,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Add a Timespan",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Add a Timespan",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": <div
              className="well"
              style={
                Object {
                  "marginTop": "1rem",
                }
              }
            >
              <Connect(HeadingFormContainer)
                cancelClick={[Function]}
              />
            </div>,
            "dimension": "height",
            "getDimensionValue": [Function],
            "in": false,
            "mountOnEnter": false,
            "timeout": 300,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(HeadingFormContainer)
                cancelClick={[Function]}
              />,
              "className": "well",
              "style": Object {
                "marginTop": "1rem",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "cancelClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": <div
              className="well"
              style={
                Object {
                  "marginTop": "1rem",
                }
              }
            >
              <Connect(TimespanFormContainer)
                cancelClick={[Function]}
                initSegment={null}
                isInitializing={true}
                timespanOpen={false}
                updateInitialize={[Function]}
              />
            </div>,
            "dimension": "height",
            "getDimensionValue": [Function],
            "in": false,
            "mountOnEnter": false,
            "timeout": 300,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(TimespanFormContainer)
                cancelClick={[Function]}
                initSegment={null}
                isInitializing={true}
                timespanOpen={false}
                updateInitialize={[Function]}
              />,
              "className": "well",
              "style": Object {
                "marginTop": "1rem",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "cancelClick": [Function],
                "initSegment": null,
                "isInitializing": true,
                "timespanOpen": false,
                "updateInitialize": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
